### PR TITLE
[no ticket][risk=no] 'Big red button' for disabling all users with notebook access on a workspace

### PR DESCRIPTION
**Pull request by als364**
Opening as draft for preliminary review while I write a bunch of tests.
---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

**Comment by als364** [comment]
argh

File: api/src/main/java/org/pmiops/workbench/api/AuthDomainController.java [hunk 1]
```diff
@@ -43,23 +46,29 @@
...
+  public ResponseEntity<Void> updateDisabledStatusForUsers(
+      UpdateDisabledStatusForUsersRequest request) {
+    final List<DbUser> dbUsers = userDao.findUsersByUsernameIn(request.getEmailList());
+    final List<DbUser> updatedDbUsers =
+        userService.setDisabledStatusForUsers(dbUsers, request.getDisabled());
+    final Map<Long, Boolean> updatedDbUsersByUserId =
+        updatedDbUsers.stream().collect(Collectors.toMap(DbUser::getUserId, DbUser::getDisabled));
+    dbUsers.forEach(
+        user -> {
+          auditAdminActions(
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/api/AuthDomainController.java:58]: Note to self: think about whether this should all be logged as one operation or many.

Have you verified in the local audit log that these are showing up?

File: api/src/main/java/org/pmiops/workbench/db/dao/UserService.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ DbUser submitDataUseAgreement(
 
   void setClusterRetryCount(int clusterRetryCount);
 
-  DbUser setDisabledStatus(Long userId, boolean disabled);
+  List<DbUser> setDisabledStatusForUsers(List<DbUser> users, boolean disabled);
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/db/dao/UserService.java:61]: nit: `isDisabled`, here and below.

File: api/src/main/resources/workbench-api.yaml [hunk 3]
```diff
@@ -5362,15 +5362,17 @@ definitions:
...
+  UpdateDisabledStatusForUsersRequest:
     type: object
     required:
-    - email
+    - emails
     properties:
-      email:
-        type: string
+      emailList:
+        type: array
```
Comment (jaycarlton) [inline api/src/main/resources/workbench-api.yaml:5371]: please add a description indicating that these are usernames and not contact emails. Better yet, rename it.

File: api/src/main/resources/workbench-api.yaml [hunk 4]
```diff
@@ -5362,15 +5362,17 @@ definitions:
...
+        items:
+          type: string
       disabled:
-        description: Set to true to disable user in auth domain, false otherwise
+        description: Set to true do disable user in auth domain, false otherwise
```
Comment (jaycarlton) [inline api/src/main/resources/workbench-api.yaml:5375]: I think this was better before.

File: ui/src/app/pages/admin/admin-workspace.tsx [hunk 5]
```diff
@@ -227,13 +235,23 @@ class AdminWorkspaceImpl extends React.Component<UrlParamsProps, State> {
...
+                <div key={i}>
+                  {userRole.email + ': ' + userRole.role}
+                </div>
+              )}
+            </FlexColumn>
+            <FlexColumn>
+              <Button style={{marginLeft: '0.5rem'}}
+                      onClick={() => {
+                        this.setState({loadingData: true});
+                        const writersAndOwners = fp.filter(userRole => userRole.role === WorkspaceAccessLevel.OWNER || userRole.role === WorkspaceAccessLevel.WRITER, collaborators);
```
Comment (jaycarlton) [inline ui/src/app/pages/admin/admin-workspace.tsx:250]: BTW, we don't yet have an approved way of doing this, but we should be able to tell you which user is actually running the cluster, so you don't have to knock out all the others.

File: api/src/main/resources/workbench-api.yaml [hunk 6]
```diff
@@ -5362,15 +5362,17 @@ definitions:
         type: array
         items:
           "$ref": "#/definitions/Profile"
-  UpdateUserDisabledRequest:
+  UpdateDisabledStatusForUsersRequest:
     type: object
     required:
-    - email
+    - emails
```
Comment (jaycarlton) [inline api/src/main/resources/workbench-api.yaml:5368]: `emails` doesn't appear to be a property. Did you mean `emailList`?

File: ui/src/app/pages/admin/admin-workspace.tsx (hunk 5 above)
Comment (als364) [reply ui/src/app/pages/admin/admin-workspace.tsx:250]: I know. This PR is meant to create a nuclear option. If the person running the cluster controls more than one user on the workspace, they'll just start the cluster with another user. Knocking out a single user should be done with the disable user script and/or the user admin.

File: ui/src/app/pages/admin/admin-workspace.tsx [hunk 7]
```diff
@@ -171,6 +172,13 @@ class AdminWorkspaceImpl extends React.Component<UrlParamsProps, State> {
     </div>;
   }
 
+  async updateDisabledStatusForUsers(usernameList: string[], disable: boolean) {
+    await authDomainApi().updateDisabledStatusForUsers(
+        {usernameList: usernameList, disabled: disable}).then(_ => {
+    });
+    this.setState({loadingData: false});
+  }
```
Comment (blrubenstein) [inline ui/src/app/pages/admin/admin-workspace.tsx:180]: Does this page currently show disabled state? if so, we may want to update the UI here.

File: ui/src/app/pages/admin/admin-workspace.tsx [hunk 8]
```diff
@@ -227,13 +235,23 @@ class AdminWorkspaceImpl extends React.Component<UrlParamsProps, State> {
...
-              <div key={i}>
-                {userRole.email + ': ' + userRole.role}
-              </div>
-            )}
-          </div>
+          <FlexRow className='collaborators' style={{marginTop: '1rem'}}>
+            <FlexColumn>
+              {collaborators.map((userRole, i) =>
+                <div key={i}>
+                  {userRole.email + ': ' + userRole.role}
```
Comment (blrubenstein) [inline ui/src/app/pages/admin/admin-workspace.tsx:242]: We may also want to show the disabled status of the users here? Just TOL.

Comment (als364) [reply ui/src/app/pages/admin/admin-workspace.tsx:242]: How about I turn the collaborators list into a table and include the disabled status there?

Comment (jaycarlton) [reply ui/src/app/pages/admin/admin-workspace.tsx:242]: +1 I also was toying with adding an indication of which one is running the cluster (maybe at the bottom), their full name, contact email, and institution (for product and security) and their database `user_id` (for DB fixes and looking up in the action audit log). I can do the backend pieces under RW-4825.

File: api/src/main/resources/workbench-api.yaml (hunk 6 above)
Comment (jaycarlton) [reply api/src/main/resources/workbench-api.yaml]: `swagger-codegen` is janky in places.

**Pull request closed without merging**